tests/common.py

```diff
@@ -212,13 +215,12 @@ def async_handle(self, intent):
 
 
 @ha.callback
-def async_fire_mqtt_message(hass, topic, payload, qos=0):
+def async_fire_mqtt_message(hass, topic, payload, qos=0, retain=False):
     """Fire the MQTT message."""
     if isinstance(payload, str):
         payload = payload.encode('utf-8')
-    dispatcher.async_dispatcher_send(
-        hass, mqtt.SIGNAL_MQTT_MESSAGE_RECEIVED, topic,
-        payload, qos)
+    dispatcher.async_dispatcher_send(hass, 'mqtt_message_received',
```

Comment:
The return of the signal dispatcher ... Even though we don't use the dispatcher for MQTT during run time, we still need some way of firing MQTT events while running tests for MQTT-based components.

One way of doing so would be to mock the paho client and use the full MQTT client code - but that would (I think) go against the principle of *unit* testing. Therefore I propose that we test the internal MQTT callbacks using `mqtt/test_init.py` and use a mock with the dispatcher for all other component tests.